larray/inout/array.py

```diff
@@ -550,9 +550,12 @@ def read_excel(filepath, sheet=0, nb_axes=None, index_col=None, fill_value=np.na
             raise TypeError("'{}' is an invalid keyword argument for this function when using the xlwings backend"
                             .format(list(kwargs.keys())[0]))
         from larray.inout.excel import open_excel
-        with open_excel(filepath) as wb:
-            return wb[sheet].load(index_col=index_col, fill_value=fill_value, sort_rows=sort_rows,
+        wb = open_excel(filepath)
+        res = wb[sheet].load(index_col=index_col, fill_value=fill_value, sort_rows=sort_rows,
                                       sort_columns=sort_columns, wide=wide)
+        if filepath != -1:
+            wb.close()
```

Comment:
explicit close is not exactly the same as the context manager, which also closes when an exception occurs.